### PR TITLE
api for todo's and trending problems

### PR DESCRIPTION
**Pull request by manav1403**

### CODE REVIEW

**Comment by manav1403** [comment]
@raj454raj, please review

File: controllers/problems.py [hunk 1]
```diff
@@ -894,16 +894,24 @@ def friends_trending():
     query = (stable.user_id.belongs(friends) | \
              stable.custom_user_id.belongs(custom_friends))
     last_submissions = trending_utilities.get_last_submissions_for_trending(query)
+     
+    # for api
+    if utilities.is_apicall():
+        return response.json(dict(problems = last_submissions))
```
Comment (raj454raj) [inline controllers/problems.py:900]:
```suggestion
        return response.json(dict(problems=last_submissions))
```

Comment (raj454raj) [reply controllers/problems.py:900]: is it last_submissions - submissions or problems? rename accordingly

File: modules/utilities.py [hunk 2]
```diff
@@ -66,8 +66,12 @@ def check_api_userauth(function):
         API Token with userauth Checking Decorator
     """
     @current.auth_jwt.allows_jwt()
+    @check_api_token
+    @current.auth.requires_login()  
     def verifier(*args, **kwargs):
-        return check_api_token(function(*args, **kwargs))
+        if current.session.user_id is None:
+            current.session.user_id = current.auth.user_id
```
Comment (raj454raj) [inline modules/utilities.py:73]: how/when is current.auth.user_id populated?

**Review by raj454raj — CHANGES_REQUESTED** [review]

File: modules/utilities.py (hunk 2 above)
Comment (manav1403) [reply modules/utilities.py:73]: it is populated by the decorator @current.auth_jwt.allows_jwt()

**Comment by manav1403** [comment]
@raj454raj  made changes, please review

File: controllers/problems.py [hunk 3]
```diff
@@ -894,16 +894,25 @@ def friends_trending():
...
 
 # ----------------------------------------------------------------------------
+@utilities.check_api_token
 def global_trending():
     from trending_utilities import draw_trending_table
     trending_problems = current.REDIS_CLIENT.get(GLOBALLY_TRENDING_PROBLEMS_CACHE_KEY)
     trending_problems = eval(trending_problems)
+    # for api
+    if utilities.is_apicall():
+        return response.json(dict(problems = trending_problems))
```
Comment (raj454raj) [inline controllers/problems.py:915]:
```suggestion
        return response.json(dict(problems=trending_problems))
```

**Review by raj454raj — CHANGES_REQUESTED** [review]

**Review by raj454raj — APPROVED** [review]

**Pull request merged**